lib/Optimizer/Transforms/ObserveAnsatz.cpp

```diff
@@ -386,13 +364,8 @@ class ObserveAnsatzPass
       return;
     }
 
-    patterns.insert<AppendMeasurements>(ctx, funcAnalysisInfo,
```

Comment:
Is it ever the case that the analysis would validly conclude not to change the FuncOp? If so, line 363 probably shouldn't signal a total failure.